### ISSUE TITLE
Remove using Neutron LBaaS v2

### ISSUE THREAD

**Issue by hidekazuna** [kubernetes-sigs/cluster-api-provider-openstack#695]
/kind feature

**Describe the solution you'd like**

Since  [Neutron LBaaS has now been retired in Train release cycle](https://docs.openstack.org/releasenotes/neutron/train.html#relnotes-15-0-0-stable-train-deprecation-notes), let's stop using it.

**Anything else you would like to add:**

OpenStack maintenance status is here: https://releases.openstack.org/

This comes from #694

**Comment by sbueringer** [comment]
I think we should wait with major breaking changes like this until 0.4.x. Not sure if we should really break compatibility in a big way like this in patch releases.

**Comment by hidekazuna** [comment]
@sbueringer I agree with you. After release-0.3 is created, we will stop using Neutron LBaaS v2 in master branch.

**Comment by jichenjc** [comment]
agree as this will break some user who is using older openstack
if we need do the change, only to specific branch..

**Comment by sbueringer** [comment]
Just have to think about when we want to start a release-0.3 branch as this means additional effort cherry-picking stuff from master to release-0.3.

Not sure if we want to create a 0.4 release just to get rid of the Neutron support or there is some kind of upcoming beta4 for ClusterAPI anyway where we can do this (I"m not really up-to-date with the project)

**Comment by hidekazuna** [comment]
I think it is better to create 0.4 release for CAPI v0.4.x (Alpha4 API) where in removing Neutron LBaaS support.

**Comment by sbueringer** [comment]
Sounds good to me

**Comment by jichenjc** [comment]
> I think it is better to create 0.4 release for CAPI v0.4.x (Alpha4 API) where in removing Neutron LBaaS support.

then we need claim 0.4 will only support openstack start from some releases 
thus we force user to upgrade their openstack version if they want to use CAPO... 
not sure it will impact our potential user

**Comment by sbueringer** [comment]
It will affect potential users, but on the other hand it's not sustainable for us to try to support versions which are sometimes years out of support. Currently the official maintained releases start with Train (https://releases.openstack.org/)

**Comment by jichenjc** [comment]
yes, pros and cons ... with current resource we have ,I agree it's reasonable to do break changes :) thanks ~

**Comment by hidekazuna** [comment]
CAPI v4.x will implement Load Balancer Provider.
https://kubernetes.slack.com/archives/CFKJB65G9/p1603728602008100

**Comment by jichenjc** [comment]
https://github.com/kubernetes-sigs/cluster-api/issues/1250 is the issue 
I think they are talk external provider which is different topic than ours ?

**Comment by sbueringer** [comment]
Nope, that is the same topic. Last time I looked at it the idea was to have a explicit concept for LoadBalancers and not just implementing the loadbalancer creation/update/deletion it in the cluster controller.

**Comment by sbueringer** [comment]
Looks like we will get rid of our last OpenStack with Neutron LBaaS v2 on Monday. So I got lucky regarding compatibility with upstream CAPO v1alpha4 :)

I would implement this
/assign
